### ISSUE TITLE
notification in workflow would be triggered twice

### ISSUE THREAD

**Issue by miyoosan** [oam-dev/kubevela#3085]
**Describe the bug**
notification in workflow would be triggered twice

**To Reproduce**

```yaml
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: whitelist-backend-test
  namespace: default
spec:
  components:
    - name: whitelist-backend-server-test
      type: webservice
      properties:
        image: <app image>
        port: 8081
      traits:
        - type: ingress-1-20
          properties:
            domain: whitelist-backend.com
            http:
              /: 8081
  workflow:
    steps:
      - name: dingtalk-pre
        type: notification
        properties:
          dingding:
            url:
              value: <dingtalk webhook url>
            message:
              msgtype: text
              text:
                content: 开始部署whitelist-backend到测试环境
      - name: application
        type: apply-component
        properties:
          component: whitelist-backend-server-test
        outputs:
          - name: app-status
            valueFrom: output.status.conditions[0].message + "部署结束"
      - name: dingtalk-post
        type: notification
        inputs:
          - from: app-status
            parameterKey: dingding.message.text.content
        properties:
          dingding:
            url:
              value: <dingtalk webhook url>
```

**Expected behavior**
notification in workflow would be triggerred once

**Screenshots**
![image](https://user-images.githubusercontent.com/25057968/149264418-ae253a9f-702b-4239-9684-4210af8abcbe.png)

**KubeVela Version**

Version: v1.2.0-beta.3
GitRevision: git-9317bb1
GolangVersion: go1.16.12

**Cluster information**
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.0", GitCommit:"ab69524f795c42094a6630298ff53f3c3ebab7f4", GitTreeState:"clean", BuildDate:"2021-12-07T18:16:20Z", GoVersion:"go1.17.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.10-r0-CCE21.11.1.B005-21.11.1.B005", GitCommit:"aa6aaf3c00ad28e5fe57be8e1b553a7f9ccb439d", GitTreeState:"clean", BuildDate:"2021-11-19T07:05:59Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}

**Additional context**

**Comment by FogDong** [comment]
Thanks for the report!
/assign

**Comment by miyoosan** [comment]
While I add a rollout trait in above application yaml file as bellow, it would go into rolloutFailed, and workflow would wait for apply-component forever.
```yaml
    traits:
       - type: rollout
          properties:
            targetSize: 3
```
![image](https://user-images.githubusercontent.com/25057968/149314596-ba91a387-b6eb-438e-b157-577750655f80.png)

Moreover,  when I change targetSize to 2, workflow will success, but it trigger twice notification.

![image](https://user-images.githubusercontent.com/25057968/149314972-2d168195-3c4a-450c-864a-1ad66b0f4b9b.png)

**Comment by FogDong** [comment]
roll out problem /cc @wangyikewxgm

**Comment by wonderflow** [comment]
I'm going to close this issue as the title described problem as been fixed in #3128 

Please create a new one for the rollout issue, thanks.
